### ISSUE TITLE
TypeError: Cannot read properties of undefined (reading 'exports')

### ISSUE THREAD

**Issue by SimonSiefke** [rollup/rollup#5043]
### Rollup Version

3.25.1

### Operating System (or Browser)

Ubuntu 23.04

### Node Version (if applicable)

20.3.0

### Link To Reproduction

https://github.com/SimonSiefke/rollup-exports-bug

### Expected Behaviour

No Error

### Actual Behaviour

```
/node_modules/rollup/dist/shared/rollup.js:17839
                        chunk.exports.add(variable);
                              ^

TypeError: Cannot read properties of undefined (reading 'exports')
    at Chunk.setUpChunkImportsAndExportsForModule (node_modules/rollup/dist/shared/rollup.js:17839:31)
    at Chunk.link (node_modules/rollup/dist/shared/rollup.js:17176:18)
    at Bundle.generateChunks (node_modules/rollup/dist/shared/rollup.js:19139:19)
    at async Bundle.generate (node_modules/rollup/dist/shared/rollup.js:19039:28)
    at async node_modules/rollup/dist/shared/rollup.js:26577:27
    at async catchUnfinishedHookActions (node_modules/rollup/dist/shared/rollup.js:25754:16)
    at async main (main.js:32:3)

Node.js v20.3.0
```

**Comment by pcfreak30** [comment]
I have just hit this as well, unsure the issue.. might have to switch bundlers as it is a blocking problem :/.

**Comment by pcfreak30** [comment]
Did a bit of digging and it appears this is either a bug or regression: PR https://github.com/rollup/rollup/pull/4952 seems to be the release/change causing this.

Release v3.20.7 is the last working version not triggering this bug.

**Comment by lukastaegert** [comment]
Here is a very minimal reproduction:

https://rollupjs.org/repl/?version=3.25.2&shareable=JTdCJTIyZXhhbXBsZSUyMiUzQW51bGwlMkMlMjJtb2R1bGVzJTIyJTNBJTVCJTdCJTIyY29kZSUyMiUzQSUyMmltcG9ydCglNUMlMjIuJTJGYS5qcyU1QyUyMiklM0IlMjIlMkMlMjJpc0VudHJ5JTIyJTNBdHJ1ZSUyQyUyMm5hbWUlMjIlM0ElMjJtYWluLmpzJTIyJTdEJTJDJTdCJTIyY29kZSUyMiUzQSUyMmV4cG9ydCUyMCU3QiUyMGIlMjAlN0QlMjBmcm9tJTIwJTVDJTIyLiUyRmIuanMlNUMlMjIlM0IlMjIlMkMlMjJpc0VudHJ5JTIyJTNBZmFsc2UlMkMlMjJuYW1lJTIyJTNBJTIyYS5qcyUyMiU3RCUyQyU3QiUyMmNvZGUlMjIlM0ElMjJleHBvcnQlMjBjb25zdCUyMGIlMjAlM0QlMjAlN0IlN0QlM0IlMjIlMkMlMjJpc0VudHJ5JTIyJTNBZmFsc2UlMkMlMjJuYW1lJTIyJTNBJTIyYi5qcyUyMiU3RCU1RCUyQyUyMm9wdGlvbnMlMjIlM0ElN0IlMjJvdXRwdXQlMjIlM0ElN0IlMjJmb3JtYXQlMjIlM0ElMjJlcyUyMiUyQyUyMmlubGluZUR5bmFtaWNJbXBvcnRzJTIyJTNBdHJ1ZSU3RCU3RCU3RA==

**Comment by lukastaegert** [comment]
Fix at #5047
